src/main/java/org/springframework/data/elasticsearch/core/ScoredEntity.java

```diff
@@ -0,0 +1,12 @@
+package org.springframework.data.elasticsearch.core;
+
+/**
+ * @author Sascha Woo
+ */
+public interface ScoredEntity {
+
+	float getScore();
+
+	void setScore(float score);
```

Comment:
Is that an interface that's supposed to be implemented by user code? If so, does it makes sense that the score can be modified? That feels quite invasive. Should we provide an annotation based approach instead, so that users can simply annotate a field in their domain class and we invoke this reflectively (via `PersistentEntity.getPropertyAccessor(…).setProperty(scoreProperty, value);` in `DefaultResultMapper? I guess, that would need `ElasticsearchPersistentEntity` to expose a score property.

Reply:
After talking to @christophstrobl, it looks like the Solr module has an [`@Score`](https://github.com/spring-projects/spring-data-solr/blob/master/src/main/java/org/springframework/data/solr/repository/Score.java) annotation that's a `@ReadOnlyProperty`. Would be cool if you could follow that design. A good starting point for the integration into the metamodel is probably [`SolrPersistentEntity`](https://github.com/spring-projects/spring-data-solr/blob/fd61a1672ac5508a80ba68db76fdd288d3e3f081/src/main/java/org/springframework/data/solr/core/mapping/SolrPersistentEntity.java#L36-L53).

Reply:
Thought exactly the same before I saw this comment ;) Will address your comments

Reply:
Feel free to ping me in [Gitter](https://gitter.im/spring-projects/spring-data) if you need help regarding the meta-model and property access.